pwiz_tools/Skyline/SettingsUI/PeptideSettingsUI.cs

```diff
@@ -343,7 +343,10 @@ private PeptideSettings ValidateNewSettings(bool showMessages)
             BackgroundProteome backgroundProteome = BackgroundProteome.NONE;
             if (!backgroundProteomeSpec.IsNone)
             {
-                backgroundProteome = new BackgroundProteome(backgroundProteomeSpec);
+                if (_peptideSettings.BackgroundProteome.EqualsSpec(backgroundProteomeSpec))

```

Comment:
I do not understand why you aren't just using regular .Equals here.

Reply:
I guess I need to add some comments. It has to do with the _cache being not truly immutable, but being used in the BackgroundProteome.Equals().